sdcm/cluster.py

```diff
@@ -3503,12 +3532,24 @@ def node_setup(self, node, **kwargs):  # pylint: disable=unused-argument
             node.remoter.run('sudo yum install screen -y')
         else:
             node.remoter.run('sudo apt-get install screen -y')
-        self.install_scylla_manager(node)
+        if self.params.get("use_mgmt", default=None):
+            self.install_scylla_manager(node, auth_token=kwargs["auth_token"])
 
-    def install_scylla_manager(self, node):
+    def install_scylla_manager(self, node, auth_token):
+        self.mgmt_auth_token = auth_token  # pylint: disable=attribute-defined-outside-init
         if self.params.get('use_mgmt', default=None):
             node.install_mgmt(scylla_repo=self.params.get('scylla_repo_m'),
-                              scylla_mgmt_repo=self.params.get('scylla_mgmt_repo'))
+                              scylla_mgmt_repo=self.params.get('scylla_mgmt_repo'), auth_token=auth_token)
+            wait.wait_for(func=self.wait_until_manager_is_up, step=20, text='Waiting until the manager client is up',
+                          timeout=300, throw_exc=True)
+
+    def wait_until_manager_is_up(self):
```

Comment:
this is not the function that does the waiting, it's just doing the check
`is_manager_up()`